archaius2-core/src/main/java/com/netflix/archaius/instrumentation/AccessMonitorUtil.java

```diff
@@ -138,15 +139,12 @@ private List<PropertyUsageEvent> createEventList(PropertyUsageEvent event) {
     }
 
     private Map<String, PropertyUsageData> getAndClearUsageMap() {
-        synchronized (propertyUsageMap) {
-            Map<String, PropertyUsageData> ret = getUsageMapImmutable();
-            propertyUsageMap.clear();
-            return ret;
-        }
+        Map<String, PropertyUsageData> map = propertyUsageMapRef.getAndSet(new ConcurrentHashMap<>());
+        return Collections.unmodifiableMap(new HashMap<>(map));
```

Comment:
No need to double wrap here. If any other thread had a reference to the map they'll release fairly soon anyway. Making a copy here will lose the updates those threads were recording. If you simply wrap it in the `unmodifiableMap` then there's a little bit more time between here and the actual flush to give them a chance to finish.

Reply:
Ah that makes sense, updated to this